qiskit_experiments/data_processing/nodes.py

```diff
@@ -555,3 +560,161 @@ def _process(self, data: np.ndarray) -> np.ndarray:
+        the number of shots and the number of circuits.
+
+        Args:
+            data: An array representing the memory.
+        """
+
+        self._n_shots = len(data[0])
+        self._n_circuits = len(data)
+
+        datum = np.array(data)
+
+        if self._validate:
+            if datum.shape != (self._n_circuits, self._n_shots):
+                raise DataProcessorError(
+                    f"The datum given to {self.__class__.__name__} does not convert "
+                    "of an array with dimension (number of circuit, number of shots)."
+                )
+
+        return data
+
+    def _reorder(self, unordered_data: np.array) -> np.array:
+        """Reorder the measured data according to the measurement sequence.
+
+        Here, is assumed that the inner loop of the measurement is done over the circuits
+        and the outer loop is done over the shots.
+        """
+        if unordered_data is None:
+            return unordered_data
+
+        order_data = []
+
+        for shot_idx in range(self._n_shots):
+            for circuit_idx in range(self._n_circuits):
+                order_data.append(unordered_data[circuit_idx][shot_idx])
+
+        return np.array(order_data)
+
+
+class RestlessToCounts(RestlessNode):
+    """Convert restless memory to counts.
+
+    This node takes as input a list of lists where the sublist is the memory of
+    each measured circuit. The sublists therefore have a length given by the
+    number of shots. This data is reordered into a one dimensional array where
+    the element at index j was the jth measured shot. This node assumes that
+    a list of circuits :code:`[circ_1, cric_2, ..., circ_m]` is measured :code:`n_shots`
+    times according to the following order:
+
+    .. parsed-literal::
+
+        [
+            circuit 1 - shot 1,
+            circuit 2 - shot 1,
+            ...
+            circuit m - shot 1,
+            circuit 1 - shot 2,
+            circuit 2 - shot 2,
+            ...
+            circuit m - shot 2,
+            circuit 1 - shot 3,
+            ...
+            circuit m - shot n,
+        ]
+
+    Once the shots have been ordered in this fashion the node compares each shot with the
+    previous shot. If they are the same then the shot corresponds to a 0, i.e. no state
+    change, and if they are different then the shot corresponds to a 1, i.e. there was
+    a state change.
+    """
+
+    def __init__(self, header: Dict[str, Any], validate: bool = True):
+        """
+        Args:
+            header: The header needed by :code:`qiskit.result.postprocess.format_counts_memory`
+                to convert the memory into a bit-string of counts. For example,
+                :code:`{"memory_slots": 1}` for a single qubit.
+            validate: If set to False the DataAction will not validate its input.
+        """
+        super().__init__(validate)
+        self._header = header
```

Comment:
I think this makes more sense
```suggestion
    def __init__(self, num_qubits: int, validate: bool = True):
        """
        Args:
            num_qubits: The number of qubits which is needed to construct the header needed
            by :code:`qiskit.result.postprocess.format_counts_memory` to convert the memory 
            into a bit-string of counts.
            validate: If set to False the DataAction will not validate its input.
        """
        super().__init__(validate)
        self._header = {"memory_slots": num_qubits}
```